_rules/document-has-instrument-to-main-ye5d6e.md

```diff
@@ -0,0 +1,620 @@
+---
+id: ye5d6e
+name: Document has an instrument to move focus to the main area of content
+rule_type: atomic
+description: |
+  This rule checks that there is an instrument to move focus to non-repeated content in the page
+accessibility_requirements:
+  wcag-technique:G1: # Adding a link at the top of each page that goes directly to the main content area
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G123: # Adding a link at the beginning of a block of repeated content to go to the end of the block
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G124: # Adding links at the top of the page to each area of the content
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - CSS styling
+  - DOM Tree
+  - Language
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+  assets:
+    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
+---
+
+## Applicability
+
+This rule applies to any [HTML web page][].
+
+## Expectation
+
+For each test target, all the following are true for the first [block of repeated content][] (in the [flat tree][]) with [perceivable content][] [at its end][at the end] which is not part of any [block of repeated content][]:
+
+- there exists at least one [keyboard actionable][] [instrument][] inside the test target to move focus [at the end][] of this [block of content][]; and
```

Comment:
This "at the end" is a little problematic. Take the following:

```
<main><h1>Main content</h1>...</main>
```

The first node in this block that meets the definition of perceivable content is the "Main content" text node, but if the mechanism goes to `main` or `h1` this should be also pass. That seems to be your intension, but I'm not sure that's actually what this is saying.